filament/backend/src/metal/MetalDriver.mm

```diff
@@ -854,22 +871,36 @@
     assert_invariant(sb->size == data.size / sizeof(SamplerDescriptor));
     auto const* const samplers = (SamplerDescriptor const*) data.buffer;
 
-#ifndef NDEBUG
-    // In debug builds, verify that all the textures in the sampler group are still alive.
+    // Verify that all the textures in the sampler group are still alive.
     // These bugs lead to memory corruption and can be difficult to track down.
     for (size_t s = 0; s < data.size / sizeof(SamplerDescriptor); s++) {
         if (!samplers[s].t) {
             continue;
         }
+        // The difference between this check and the one below is that in release, we do this for
+        // only a set number of recently freed textures, while the debug check is exhaustive.
+        auto* metalTexture = handle_cast<MetalTexture>(samplers[s].t);
+        if (UTILS_UNLIKELY(metalTexture->isTerminated())) {
+            NSString* reason = [NSString stringWithFormat:
+                    @"Filament Metal texture use after free, sampler "
+                    @"group = %s, texture index = %zu",
+                    sb->debugName.c_str_safe(), s];
+            NSException* useAfterFreeException =
+                    [NSException exceptionWithName:@"MetalTextureUseAfterFree"
+                                            reason:reason
+                                          userInfo:nil];
+            [useAfterFreeException raise];
+        }
```

Comment:
this code looks identical to the one in `finalizeSamplerGroup` ca we move it in a "checkUseAfterFree" method?